src/lib/trace-ir/field-class.c

```diff
@@ -669,6 +671,23 @@ add_mapping_to_enumeration_field_class(struct bt_field_class *fc,
 	BT_LIB_LOGD("Added mapping to enumeration field class: "
 		"%![fc-]+F, label=\"%s\"", fc, label);
 
+	// See if this remains a bit field enum field
+	if (enum_fc->is_bit_field_enum) {
+		if (range_set->ranges->len != 1) {
+			enum_fc->is_bit_field_enum = false;
+		} else {
+			const struct bt_integer_range *range = (const void *)
+				BT_INTEGER_RANGE_SET_RANGE_AT_INDEX(
+					range_set, 0);
+			if ((range->lower.i != range->upper.i) ||
```

Comment:
This excludes enums that contain at least one range which contains more than a single value. (perhaps a bit of comments ?)